src/DotNetProjectFile.Analyzers/Analyzers/MsBuild/ThirdPartyLicenseResolver.cs

```diff
@@ -24,50 +24,73 @@ protected override void Register(ProjectFileAnalysisContext context)
             .Where(p => p.Include is { Length: > 0 })
             .ToArray();
 
+        var queue = new Queue<Dependency>();
+        var done = new HashSet<PackageVersionInfo>();
+
         foreach (var reference in context.File.ItemGroups.SelectMany(g => g.Children)
             .OfType<PackageReferenceBase>()
-            .Where(r => r.Version is { Length: > 0 }))
+            .Where(r => r is { Info.Version.Length: > 0 } && done.Add(r.Info)))
         {
-            Report(reference, projectLicense, licenses, context);
+            queue.Enqueue(new(reference, reference.Info));
```

Comment:
shouldn't this also be guarded by a `if(done.Add(...))` ? Otherwise you might still see some duplicates

Reply:
The `.Where` clause contains that constraint, doesn't it?